webapp-client/src/errorify.js

```diff
@@ -0,0 +1,30 @@
+import ErrorMessage from './component/ErrorMessage';
+import React from 'react';
+
+/**
+ * allowErrorMessage modifies the given component to give it a boolean state
+ * variable called hasError. Along with this, the methods showErrorMessage,
+ * rendorErrorMessage and clearErrorMessage are added. These allow the component
+ * to easily make use of the ErrorMessage component.
+ * @param {React.Component} component - A react component
+ */
+export default function allowErrorMessage(component) {
```

Comment:
Where is this function going to be used?

Reply:
You can see how it will be used in #30. It was split off to be merged into master since Cynthia and I will both be using it in a few different places :D